### PR TITLE
Add Test Tool and QA Tools

### PR DESCRIPTION
**Pull request by maxmeyer**
Hi there,
as discussed in #35 here's a PR for basic stuff to setup testing. I also added `simplecov`, `rubocop` and `inch` for qa. I also added some more rake tasks and organize it a bit differently. Hope this was ok.

Next QA-step would be to let `rubocop` fix most offenses automatically. It does a decent job and is non-destrutive from my experience. There are ~ 200 offenses not fix which might be discussed first and are non-trival to fix.

### CODE REVIEW

**Comment by maxmeyer** [comment]
I also added `activesupport` to `test` and `development` group to make use of `strip_heredoc` which is quite handy if you have to handle long indented strings. It removes all leading whitespace which makes it easy to use this in tests.

```
<<-EOS.strip_heredoc
  def blub
    "hello world"
  end
EOS

**Comment by jasonlong** [comment]
Thanks for working on this @maxmeyer. I'm excited to get a proper test suite in place.

Your PRs go a bit deeper than my Ruby chops are comfortable with, so I'd :heart: to get some input from some other folks as well. 

@andrew @ttaylorr @tarebyte @btmills Would any of you guys be willing to help review this as well as #39, #40, and #41?

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+SimpleCov.start
+
+# Pull in all of the gems including those in the `test` group
+require 'bundler'
+Bundler.require :default, :test, :development
+
+require 'geo_pattern'
+
+# Loading support files
+Dir.glob(::File.expand_path('../support/*.rb', __FILE__)).each { |f| require_relative f }
```
Comment (ttaylorr) [inline spec/spec_helper.rb]: We could probably factor out a `#require_in_dir(directory)` method that does this and use that instead.

**Comment by ttaylorr** [comment]
Thanks @jasonlong, happy to.  This PR looks fine to me, and I like the direction it's headed in.  I had a few minor, style-related concerns, but besides that everything else looks good.

**Comment by maxmeyer** [comment]
Better? :-)

File: lib/geo_pattern/helpers.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module GeoPattern
+  module Helpers
+    def require_files_matching_pattern(pattern)
```
Comment (ttaylorr) [inline lib/geo_pattern/helpers.rb]: :+1:

Comment (maxmeyer) [inline lib/geo_pattern/helpers.rb]: Fine. :-)

**Comment by ttaylorr** [comment]
Thanks!  I'm happy with this PR, lgtm.

/to @jasonlong

**Comment by maxmeyer** [comment]
Fixed merge conflict.

**Comment by maxmeyer** [comment]
And rebased....

**Comment by jasonlong** [comment]
2 down!

**Comment by maxmeyer** [comment]
Thanks

**Pull request merged**